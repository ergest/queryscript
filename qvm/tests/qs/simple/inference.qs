export let users = load('users.json');
export let events = load('events.json');

select * from users order by id;
select * from events order by ts;
