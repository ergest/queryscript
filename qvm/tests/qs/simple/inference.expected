{
    "compile_errors": [],
    "decls": {
        "let events": External<[{
        	description Utf8,
        	ts Utf8,
        	user_id Int64,
        }]>,
        "let users": External<[{
        	active Boolean,
        	id Int64,
        	name Utf8,
        	org_id Int64,
        }]>,
    },
    "queries": [
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "active",
                                type_: Atom(
                                    Boolean,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "id",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "name",
                                type_: Atom(
                                    Utf8,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "org_id",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| active | id | name | org_id |\n|--------|----|------|--------|\n| true   | 1  | Foo  | 1      |\n| false  | 2  | Bar  | 1      |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "description",
                                type_: Atom(
                                    Utf8,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "ts",
                                type_: Atom(
                                    Utf8,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "user_id",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| description | ts         | user_id |\n|-------------|------------|---------|\n| Loren Ipsum | 2020-01-01 | 1       |\n| Foo Bar     | 2020-01-02 | 1       |\n| Bing Baz    | 2020-01-03 | 2       |\n| Woo Hoo     | 2020-01-04 | 2       |",
            },
        ),
    ],
}